src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -195,13 +195,11 @@ private function createAuthorization($config, ContainerBuilder $container)
         ));
 
         foreach ($config['access_control'] as $access) {
-            $matcher = $this->createRequestMatcher(
-                $container,
-                $access['path'],
-                $access['host'],
-                $access['methods'],
-                $access['ips']
-            );
+            if (null !== $access['matcher']) {
+                $matcher = new Reference($access['matcher']);
```

Comment:
Im not sure about the "right" approach here. Perhaps we should talk config/API first.

But we could do `matcher|<combination of other keys>`.

DX-wise i think `- { route: foo, roles: [] }` is best and probably same for firewalls? (using an array of routes).

As goes for `- { expression: foo }`. But that could work counter-intuitive with `allow_if`.